eth/abc.py

```diff
@@ -2490,14 +2541,14 @@ def make_receipt(self,
     # Mining
     #
     @abstractmethod
-    def import_block(self, block: BlockAPI) -> BlockAPI:
+    def import_block(self, block: BlockAPI) -> BlockEvalResult:
```

Comment:
This signature change (along with the `mine_block` sig changes) made this PR a lot larger. It seems like a lot of callers don't really care about the witness! To make things easier for those callers, an alternative might be for this method to stay the same and methods added instead: `import_block_returning_witness_index` and `mine_block_returning_witness_index`.